pkg/secrethub/iterator.go

```diff
@@ -0,0 +1,40 @@
+package secrethub
+
+import (
+	"github.com/secrethub/secrethub-go/internals/errio"
+)
+
+// Errors
+var (
+	IteratorDone = errio.Namespace("iterator").Code("done").Error("there are no more items left")
+)
+
+type paginator interface {
+	Next() ([]interface{}, error)
+}
+
+type iterator struct {
```

Comment:
One thing to be careful about with these types of abstractions is how this works if multiple goroutines call `iterator.Next()`. Seeing as we're exposing this as a library, I feel like the iterator should be able to handle that nicely, e.g. with an internal mutex?

What are your thoughts on this subject?